lib/path.js

```diff
@@ -290,16 +297,15 @@ const win32 = {
           } else {
             rootEnd = 1;
           }
-        } else if ((code >= CHAR_UPPERCASE_A && code <= CHAR_UPPERCASE_Z) ||
-                   (code >= CHAR_LOWERCASE_A && code <= CHAR_LOWERCASE_Z)) {
+        } else if (isWindowsDeviceRoot(code)) {
           // Possible device root
 
           if (path.charCodeAt(1) === CHAR_COLON) {
             device = path.slice(0, 2);
             rootEnd = 2;
             if (len > 2) {
               code = path.charCodeAt(2);
-              if (code === CHAR_FORWARD_SLASH || code === CHAR_BACKWARD_SLASH) {
+              if (isPathSeparator(code)) {
```

Comment:
It should be fine to inline `path.charCodeAt(2)`

Reply:
Ok, this applies to a lot of places here ;-)

Reply:
@BridgeAR there are too many places like this. I think it will be better to fix it in a another pull request

Reply:
Hm, I personally think this would actually be the right place to do so. The reason is that it touches the same lines. @apapirovski what do you think?

Reply:
`code` seems to mostly be there to avoid constantly calling path.charCodeAt(x) within the same conditional but now that these are a function, it seems like their purpose is gone. I would say that this could definitely be done in this PR even though it adds a bunch of work.

Reply:
@BridgeAR ok, I was just worried that it might affect some logic here. Nonetheless, I'll fix it

Reply:
@daynin it'll probably take some effort to validate all the places where it's safe to do so. Would suggest doing it as a separate commit. Should make reviews easier.

Reply:
@apaprocki ok, sure

Reply:
It should be relatively straight forward. Try to remove the from the most inner function and go to the outer one. I checked it for one big function and as far as I saw it was possible to remove all of them besides a single one that should switch to be const.

Reply:
@apapirovski @BridgeAR fixed